src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java

```diff
@@ -191,135 +270,206 @@
- * </p>
- * <p>
- * Examples of Checkstyle checks configuration:
- * </p>
- * <pre>
- * &lt;module name="RegexpSinglelineJava"&gt;
- *   &lt;property name="id" value="ignore"/&gt;
- *   &lt;property name="format" value="^.*@Ignore\s*$"/&gt;
- *   &lt;property name="message" value="@Ignore should have a reason."/&gt;
- * &lt;/module&gt;
- *
- * &lt;module name="RegexpSinglelineJava"&gt;
- *   &lt;property name="id" value="systemout"/&gt;
- *   &lt;property name="format" value="^.*System\.(out|err).*$"/&gt;
- *   &lt;property name="message" value="Don't use System.out/err, use SLF4J instead."/&gt;
- * &lt;/module&gt;
- * </pre>
- * <p>
- * Example of SuppressionCommentFilter configuration (checkFormat which is set
- * to '$1' points that ID of the checks is in the first group of offCommentFormat
- * and onCommentFormat regular expressions):
+ * In the config of SuppressionCommentFilter, checkFormat is set to '$1' which points
+ * to the ID written in the offCommentFormat and onCommentFormat. Config for such a
+ * case is written below:
  * </p>
  * <pre>
  * &lt;module name="SuppressionCommentFilter"&gt;
- *   &lt;property name="offCommentFormat" value="CSOFF (\w+) \(\w+\)"/&gt;
+ *   &lt;property name="offCommentFormat" value="CSOFF (\w+)"/&gt;
  *   &lt;property name="onCommentFormat" value="CSON (\w+)"/&gt;
  *   &lt;property name="idFormat" value="$1"/&gt;
  * &lt;/module&gt;
+ * &lt;module name="MemberName"&gt;
+ *   &lt;property name="id" value="MemberID"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="ConstantName"&gt;
+ *   &lt;property name="id" value="ConstantID"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="IllegalCatch"&gt;
+ *   &lt;property name="id" value="IllegalID"/&gt;
+ * &lt;/module&gt;
  * </pre>
  * <pre>
- * // CSOFF ignore (test has not been implemented yet)
- * &#64;Ignore // should NOT fail RegexpSinglelineJava
- * &#64;Test
- * public void testMethod() { }
- * // CSON ignore
- *
- * // CSOFF systemout (debug)
- * public static void foo() {
- *   System.out.println("Debug info."); // should NOT fail RegexpSinglelineJava
+ * class InputSuppressionCommentFilter
+ * {
+ *  int VAR1; // violation , Name 'VAR1' must match pattern '^[a-z][a-zA-Z0-9]*$'
+ *
+ *  //CSOFF MemberID
+ *  int VAR2; // suppressed violation
+ *  //CSON: MemberID
+ *
+ *  public static final int var3;
+ *  // violation above , Name 'var3' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'
+ *
+ *  //CSOFF ConstantID
+ *  public static final int var4; // suppressed violation
+ *  //CSON ConstantID
+ *
+ *  public void method1() {
+ *
+ *  try{
+ *
+ *  }catch(Exception ex){ // violation , Catching 'Exception' is not allowed
+ *
+ *  }
+ *
+ *  //CSOFF IllegalID
+ *  try{
+ *
+ *  }catch(Exception ex){ // suppressed violation
+ *
+ *  }catch(Error err){ // suppressed violation
+ *
+ *  }
+ *  //CSON IllegalID
+ *  }
  * }
- * // CSON systemout
  * </pre>
  * <p>
  * Example of how to configure the check to suppress more than one checks.
```

Comment:
`Example of how to configure the check to suppress Checks by name defined in comment.`

Reply:
Done